test/ParameterStoreBuildWrapperPluginTest.groovy

```diff
@@ -25,6 +33,154 @@ class ParameterStoreBuildWrapperPluginTest {
+            configureJenkins(repoName: repoName, organization: organization)
+
+            TerraformEnvironmentStage stage         = mock(TerraformEnvironmentStage.class)
+            ParameterStoreBuildWrapperPlugin plugin = spy(new ParameterStoreBuildWrapperPlugin())
+
+            doReturn(environment).when(stage).getEnvironment()
+            doReturn(apo).when(plugin).getEnvironmentParameterOptions(environment)
+            doReturn(firstClosure).when(plugin).addParameterStoreBuildWrapper(gp)
+            doReturn(secondClosure).when(plugin).addParameterStoreBuildWrapper(apo)
+
+            plugin.withGlobalParameter(path)
+            plugin.apply(stage)
+
+            verify(stage, times(2)).decorate(anyString(), eq(firstClosure))
+            verify(stage, times(2)).decorate(anyString(), eq(secondClosure))
+        }
+
+    }
+
+    public class GetEnvironmentParameterOptions {
+        @After
+        public void reset() {
+            Jenkinsfile.instance = null
+            ParameterStoreBuildWrapperPlugin.reset()
+        }
+
+        private configureJenkins(Map config = [:]) {
+            Jenkinsfile.instance = mock(Jenkinsfile.class)
+            when(Jenkinsfile.instance.getStandardizedRepoSlug()).thenReturn(config.repoSlug)
+            when(Jenkinsfile.instance.getRepoName()).thenReturn(config.repoName ?: 'repo')
+            when(Jenkinsfile.instance.getOrganization()).thenReturn(config.organization ?: 'org')
+            when(Jenkinsfile.instance.getEnv()).thenReturn(config.env ?: [:])
+        }
+
+        @Test
+        void validPathBasedOnEnvironment() {
+            String organization = "MyOrg"
+            String repoName     = "MyRepo"
+            String environment  = "qa"
+            Map expected        = [path: "/${organization}/${repoName}/${environment}/".toString()]
+
+            configureJenkins(repoName: repoName, organization: organization)
+            ParameterStoreBuildWrapperPlugin plugin = spy(new ParameterStoreBuildWrapperPlugin())
+            doReturn(expected.path.toString()).when(plugin).pathForEnvironment(environment)
+
+            Map actual = plugin.getEnvironmentParameterOptions(environment)
+
+            assertEquals(expected.path, actual.path)
+        }
+
+        @Test
+        void validCredentialsBasedOnEnvironment() {
+            String organization = "MyOrg"
+            String repoName     = "MyRepo"
+            String environment  = "qa"
+            Map expected        = [credentialsId: "${environment.toUpperCase()}_PARAMETER_STORE_ACCESS"]
+
+            configureJenkins(repoName: repoName, organization: organization)
+            ParameterStoreBuildWrapperPlugin plugin = spy(new ParameterStoreBuildWrapperPlugin())
+            doReturn(expected.path.toString()).when(plugin).pathForEnvironment(environment)
+
+            Map actual = plugin.getEnvironmentParameterOptions(environment)
+
+            assertEquals(expected.credentialsId, actual.credentialsId)
+        }
```

Comment:
The same simplifications above can be made here.

Reply:
done